src/core/net/ip6.cpp

```diff
@@ -497,6 +501,8 @@ Error Ip6::SendDatagram(Message &aMessage, MessageInfo &aMessageInfo, uint8_t aI
     }
     else
     {
+        aMessage.SetAllowLoopbackToHost(true);
+        aMessage.SetOrigin(Message::kOriginHostTrusted);
```

Comment:
Can we assume the caller set these already on `aMessage`? (which may be the default we set when we allocate the message)?